src/opt/Optimizer/Optimizer.hs

```diff
@@ -128,3 +129,28 @@ dropBorrowBlocks = extend dropBorrowBlock
            ,decls = [([VarNoType name], target)]
            ,body}
       dropBorrowBlock e = e
+
+forwardExistingFuture = extend forwardExistingFuture'
+  where
+    forwardExistingFuture' e@(Forward{emeta, forwardExpr=arg@VarAccess{}})=Forward{emeta=emeta', forwardExpr=newExpr}
+      where
+         forwardExpr = arg
```

Comment:
This line is redundant. Both `forwardExpr` and `arg` already refer to the same thing.